### DISCUSSION TITLE
Version parts

### DISCUSSION THREAD

**Discussion by aowss** [OAI/OpenAPI-Specification#2645] [Q&A]
Are all parts of the [`openapi`](https://github.com/OAI/OpenAPI-Specification/blob/v3.1.1-dev/versions/3.1.1.md#versions) field mandatory or can i have a value that is just `3.1`?

**Comment by MikeRalphson** [comment]
The major, minor and patch numbers of the `openapi` field are mandatory (the value is a semantic version number, not a semver matching expression), but the patch version SHOULD be ignored by tooling, so `3.1.0`, `3.1.1` and `3.1.999` are treated equivalently.

You can also have pre-release information following a dash, such as `3.2.0-rc0`. This is obviously optional.

**Reply by char0n** [reply]
Hi @MikeRalphson can you please explain the meaning of: "the value is a semantic version number, not a semver matching expression" ? Thanks!

**Reply by MikeRalphson** [reply]
@char0n sure, back at my laptop now rather than on my phone so finding links is easier. The `openapi` field value is a [normal version number](https://semver.org/#spec-item-2) according to the [semver spec](https://semver.org/), so MUST contain all three components (major,minor,patch). 

By "semver matching expression" I meant something like [npmjs's "ranges"](https://docs.npmjs.com/cli/v6/using-npm/semver#ranges). Here "3.1" would be treated as an ["X-Range"](https://docs.npmjs.com/cli/v6/using-npm/semver#x-ranges-12x-1x-12-) - so drawing the distinction that "3.1" would NOT be a valid value for the `openapi` field.

**Reply by char0n** [reply]
Thank for explanation! Though the `openapi` field format is compliant with semver spec, the actual meaning of the number is IMHO not. As OAS 3.1.0 technically contains multiple breaking changes we would be in OAS 4.0.0 instead of OAS 3.1.0.

**Reply by MikeRalphson** [reply]
@char0n you are absolutely right, we did drop semver in v3.1.0 and I should have made that clear. We still subscribe to the definitions of **major**, **minor**, and **patch** versions/releases, but we class some theoretically-breaking changes as minor, because of their limited impact.

**Reply by aowss** [reply]
thanks for your prompt and detailed version.
Should a link to [normal version number](https://semver.org/#spec-item-2) be added to the spec to make that clearer ?

**Reply by MikeRalphson** [reply]
@aowss Probably because semver was officially dropped in v3.1, linking to parts of that spec again might be confusing. I think we'd totally welcome a PR to the spec. clarifying that the major, minor and patch numbers of the `openapi` field are all mandatory and that there is the optional pre-release information, as we do seem to have lost that specific guidance.

**Comment by handrews** [comment]
Since we are sorting out our release scale policies in the following issues, I'm going to close this discussion in favor of them:

* #3528 
* #3529
